### PR TITLE
customer Orders query shipment comments created_at is returned in +0 timezone not in store configured timezone

### LINKED ISSUE #36947: customer Orders query shipment comments created_at is returned in +0 timezone not in store configured timezone
### Preconditions and environment

Magento 2.4.x

### Steps to reproduce

1. Configure your store with a timezone different from +0
2. Create a Shipment with a comment
3. Verify the comment created_at in the normal view phtml in the website
4. Call customer Orders query and check created_at field from shipment comment

### Expected result

The expected result is that the time from both view and query be the same and in our configured timezone

### Actual result

The actual result is that the time from view is right in our timezone, and in query it is in +0 timezone

### Additional information

In view phtml the created_at goes through a formatDate function
path: Magento_Sales::order/comments.phtml

In query processing, it does not
path: Magento\SalesGraphQl\Model\Resolver\Shipments line 65

### Release note

_No response_

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by Kannakiraj123**
### Description (*)
This pull request (PR) will be change created_at date to local timezone format date in customer order query
   

### Related Pull Requests
 I added function to change created_at date to local timezone format date.

### Fixed Issues (if relevant)

1. Fixes #36947 

### Manual testing scenarios (*)
1. Create order
2. Call customer Orders query and check created_at, order_date  field from orders
3. Result is that the time from both admin order view page and query be the same and in our configured timezone

Screenshot 2023-06-20 at 5 14 57 PM

Screenshot 2023-06-20 at 5 14 41 PM

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Kannakiraj123** [comment]
@magento run all tests

**Comment by Kannakiraj123** [comment]
@magento run all tests

**Comment by Kannakiraj123** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Integration Tests, WebAPI Tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @Kannakiraj123,

Thanks for the contribution!

As part of the [DOD](https://developer.adobe.com/commerce/contributor/guides/code-contributions/definition-of-done/#definition-of-done-1), please cover the changes with some automated tests.

It seems the build failures are due to PR changes, 2 consecutive build runs generate the same failures for eg:
Integration Test Failures:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37642/57cdd3994f3d01648fd8b948238415f4/Integration/console-error-logs.html

https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37642/9bc23ef2e0b2f9f08101a252a1280025/Integration/console-error-logs.html

WebApi Tests:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37642/cfba89fccf9b7e6a7cfd15e02df92968/WebApi/console-error-logs.html

https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37642/9a1ee6305a17424e3c5767e83bdd0af3/WebApi/console-error-logs.html

Please fix those as well

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B, Integration Tests, WebAPI Tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE, Functional Tests B2B, WebAPI Tests

**Comment by engcom-Charlie** [comment]
Hi @Kannakiraj123,

Thank you for your contribution!

It seems that you are working on this PR, are you going o take care of build failures as well as automation test coverage. Please confirm once.

Thank you!

**Comment by Kannakiraj123** [comment]
@magento run WebAPI Tests

**Comment by Kannakiraj123** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run WebAPI Tests,Functional Tests EE,Functional Tests CE

File: app/code/Magento/SalesGraphQl/Model/Formatter/Order.php [hunk 1]
```diff
@@ -26,16 +27,24 @@ class Order
...
     /**
      * @param OrderAddress $orderAddress
      * @param OrderPayments $orderPayments
+     * @param Converter $converter
      */
     public function __construct(
         OrderAddress $orderAddress,
-        OrderPayments $orderPayments
+        OrderPayments $orderPayments,
+        Converter $converter
```
Comment (engcom-Hotel) [inline app/code/Magento/SalesGraphQl/Model/Formatter/Order.php]: Adding new parameter in this way is backward incompatible, please go through with the below document for reference:

https://developer.adobe.com/commerce/contributor/guides/code-contributions/backward-compatibility-policy/#adding-a-constructor-parameter

Please fix the same in other places as well.

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @Kannakiraj123,

Thanks for the contribution!

Please have a look into the review points.

Thanks

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Integration Tests,Functional Tests CE

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Review by engcom-Hotel — APPROVED** [review]

**Comment by engcom-Delta** [comment]
Hi @Kannakiraj123, @engcom-Hotel ,

Thanks for the collaboration & contribution!

### :x: QA not Passed    
                                    
Customer Orders query shipment comments created_at is returned in +0 timezone not in store configured timezone

**Manual Scenario Steps**

Configure your store with a timezone different from +0
Create a Shipment with a comment
Verify the comment created_at in the normal view phtml in the website
Call customer Orders query and check created_at field from shipment comment

**:heavy_check_mark: Expected result After Fix** 

 The time from both view and query be the same and in our configured timezone

**:x: Actual result After Fix** 

Timezone
Screenshot 2023-10-18 at 7 22 09 PM

Order detail in admin 
Screenshot 2023-10-18 at 6 55 19 PM

Order in Graphql
Screenshot 2023-10-18 at 7 03 19 PM

**Review by engcom-Delta — CHANGES_REQUESTED** [review]
The time from view is right in our timezone, and in query it is in +0 timezone

**Comment by engcom-Delta** [comment]
Hi @Kannakiraj123,
Thanks for your contribution. 

As per the issue linked to this https://github.com/magento/magento2/issues/36947 we are facing multiple issues which were not identified earlier, Please check comment added by the contributor on the issue :  https://github.com/magento/magento2/issues/36947#issuecomment-1773333104.

Steps to reproduce followed to test the PR as follows.

Configure store with a timezone different from +0

new

Create a Shipment with a comment

new1

Verify the comment created_at in the normal view phtml in the website

new2

Call customer Orders query and check created_at field from shipment comment

new3

We found out that we are facing issue with everywhere where timezone is present and after taking the PR changes we were not able to get the timezone fixed in any scenario.

Let us know the scenarios for which the PR is made as there are some conflicts present related to the PR and the linked issue and confirm if we have keep the PR on hold

Thanks,

**Comment by engcom-Delta** [comment]
Due to the below additional issue and conflicts present in the linked issue and the PR we are moving it to on hold status

https://github.com/magento/magento2/pull/37642#issuecomment-1793107311

**Comment by engcom-Delta** [comment]
Hi @Kannakiraj123 , @engcom-Hotel 

Thanks for the collaboration & contribution!

### :x: QA not Passed

Customer Orders query shipment comments created_at is returned in +0 timezone still not in store configured timezone

**Manual Scenario Steps**

1.  Configure your store with a timezone different from +0
2. Create a Shipment with a comment
3. Verify the comment created_at in the normal view phtml in the website
4. Call customer Orders query and check created_at field from shipment comment

**:heavy_check_mark: Expected result After Fix** 

Customer Orders query shipment comments created_at is returned in +0 timezone should be in store configured timezone

**:x: Actual result After Fix** 

Configured store with a timezone different from +0

Screenshot 2023-11-14 at 8 13 08 PM

Created a Shipment with a comment

Screenshot 2023-11-14 at 8 21 48 PM

Verify the comment created_at in the normal view phtml in the website

Screenshot 2023-11-14 at 8 22 06 PM

Screenshot 2023-11-14 at 8 22 26 PM

Screenshot 2023-11-14 at 8 23 14 PM

Called customer Orders query and check created_at field from shipment comment

Screenshot 2023-11-14 at 8 25 00 PM


Hence moving it to Changes requested.

**Review by engcom-Delta — CHANGES_REQUESTED** [review]
The time from view is right in our timezone, and in query it is in +0 timezone

**Comment by engcom-Delta** [comment]
As the basic functionality for the PR is not working , moving this to on hold status. PFB the comment

https://github.com/magento/magento2/pull/37642#issuecomment-1810409304

**Pull request closed without merging**